search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,275 @@
+ */
+async function updateSearchResultsForTopic(query) {
+  let countriesData = [];
+  await deleteAncientResults();
+
+  // Note: Can't use forEach with await.
+  for (let i = 0; i < json.length; i++) { // When testing, make i < 3 countries. 
+    // 100 queries per minute limit for Custom Search API. Pause to prevent
+    // surpassing limit.
+    if (i !== 0 && i % 100 === 0) {
+      await new Promise(resolve => setTimeout(resolve, 60000));
+    }
+    let countryData = [];
+    // Update countryData within the functions called.
+    const avgCountrySentimentScore = await getSearchResultsForCountryFromAPI(
+        "country" + json[i].id, query, countryData);
+    // TODO(ntarn): Remove when done with sentiment chart feature debugging.
+    console.log('ntarn debug country: ' + json[i].id + ' averageSentiment: ' +
+        avgCountrySentimentScore);
+    countriesData.push({
+      country: json[i].id,
+      averageSentiment: avgCountrySentimentScore,
+      results: countryData,
+    });
+  }
+  addTopicToDatastore(query, countriesData);
+}
+
+/** 
+ * Gets top 10 search results for specified country and query from Custom
+ * Search API and passes them to be saved in countryData object.
+ * @param {string} countryCode 2 letter country code for search results to be
+ *     written in.
+ * @param {string} query Search query.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ * @return {number} The average sentiment score propagated from the
+ * formatResults method.
+ */
+async function getSearchResultsForCountryFromAPI(countryCode, query, countryData) {
+  let response =
+    await fetch('https://www.googleapis.com/customsearch/v1?key=AIzaSyDszWv1aGP7Q1uOt74CqBpx87KpkhDR6Io&cx=017187910465527070415:o5pur9drtw0&q=' + query + '&cr=' + countryCode + '&num=10&safe=active&dateRestrict=d1&fields=items(title,snippet,htmlTitle,link)');
+  let searchResults = await response.json();
+  return await formatCountryResults(searchResults, countryData);
+}
+
+/**
+ * Formats the current results given the results JSON obtained from theAPI.
+ * @param {Object} searchResultsJson Object with information for top 10 search
+ *     results.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function formatCountryResults(searchResultsJson, countryData) {
+  // Parse the JSON string and pass each search result to add to the
+  // countryData object.
+  let currentSearchResults = searchResultsJson.items;
+  try {
+    let avg = 0;
+    if (currentSearchResults == undefined) {
+      return 0;
+    } else {
+      for (let i = 0; i < currentSearchResults.length; i++) {
+        let formattedResults =
+            await formatSearchResults(currentSearchResults[i]);
+        countryData.push(formattedResults);
+        avg += formattedResults.score;
+      }
+      return avg / currentSearchResults.length;
+    }
+  } catch (err) { // Occurs when no search results for that country and topic.
```

Comment:
1 this err is too broad, is there a specific error that is thrown?
2 sounds like this error is thrown by formatSearchResults? If so, can you wrap the try catch just around the formatSearchResults line instead of wrapping it with that big scope?

Reply:
The error was actually caused when trying to divide by 0(when currentSearchResults.length was 0). Handled this with an if block instead.